modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java

```diff
@@ -261,38 +309,151 @@ private PredicateExpressionVisitorImpl(
 		Map<String, String> lambdaVariableExpressionFieldNames,
 		long objectDefinitionId,
 		ObjectFieldBusinessTypeRegistry objectFieldBusinessTypeRegistry,
-		ObjectFieldLocalService objectFieldLocalService) {
+		ObjectFieldLocalService objectFieldLocalService,
+		ObjectRelatedModelsPredicateProviderRegistry
+			objectRelatedModelsPredicateProviderRegistry) {
 
-		_entityModel = entityModel;
+		_objectDefinitionsEntityModelMap.put(objectDefinitionId, entityModel);
 		_lambdaVariableExpressionFieldNames =
 			lambdaVariableExpressionFieldNames;
 		_objectDefinitionId = objectDefinitionId;
 		_objectFieldBusinessTypeRegistry = objectFieldBusinessTypeRegistry;
 		_objectFieldLocalService = objectFieldLocalService;
+		_objectRelatedModelsPredicateProviderRegistry =
+			objectRelatedModelsPredicateProviderRegistry;
 	}
 
 	private Predicate _contains(Object fieldName, Object fieldValue) {
-		Column<?, Object> column = _getColumn(fieldName);
+		Column<?, Object> column = _getColumn(fieldName, _objectDefinitionId);
 
 		return column.like(
-			StringPool.PERCENT + _getValue(fieldName, fieldValue) +
-				StringPool.PERCENT);
+			StringPool.PERCENT +
+				_getValue(_objectDefinitionId, fieldName, fieldValue) +
+					StringPool.PERCENT);
 	}
 
-	private Column<?, Object> _getColumn(Object fieldName) {
-		EntityField entityField = _getEntityField(fieldName);
+	private EntityModel _createEntityModel(long objectDefinitionId) {
+		try {
+			return new ObjectEntryEntityModel(
+				ObjectDefinitionLocalServiceUtil.getObjectDefinition(
+					objectDefinitionId),
+				_objectFieldLocalService.getObjectFields(objectDefinitionId));
+		}
+		catch (Exception exception) {
+			_logExceptionInDebugMode(exception);
+
+			return new ObjectEntryEntityModel(Collections.emptyList());
+		}
+	}
+
+	private ObjectRelationship _fetchObjectRelationship(
```

Comment:
Arrrg, I think this method with try catch is a bit ugly. We should use fetch instead.

I talked to Sergio about it, because as I guess you realised, there is a fetch method that does a different action, based in the objectDefinitionId1.

We talked about renaming the current `fetchObjectRelationshipByObjectDefinitionId` to `fetchObjectRelationshipByObjectDefinitionId1` and then, create a new `fetchObjectRelationshipByObjectDefinitionId` method with the same action that the _get_ one but without throwing any exception, but I guess his pull request is still on development (or nothing has been done yet maybe).

Anyway, for now it is OK, but I guess we should do it in the future

Reply:
Yep, I know it and I saw that new method in Sergio's PR. I did it this way to avoid conflicts between PRs. I hope we remember to change this in the future and rely on the service this specific logic. (Maybe I should create a debt ticket to this in case we merge it)

Reply:
Or create a ticket or write it down in a paper and paste in into your computer :joy: